### PR TITLE
fix: Add longer sleep in end_to_end_cases::debug::build_catalog

### PR DESCRIPTION
**Pull request by alamb**
Closes https://github.com/influxdata/influxdb_iox/issues/8203 (I hope)

@savage-engineer  added a retry loop as part of the fix for https://github.com/influxdata/influxdb_iox/issues/8200, but I believe on the CI machines the loop is not long enough (it tries to query the server up to 10x a second for 2 seconds)

This PR increases the wait time to 5 seconds and decreases the poll interval to 2x a second for up to 5 seconds

### CODE REVIEW

File: influxdb_iox/tests/end_to_end_cases/debug.rs [hunk 1]
```diff
@@ -105,20 +105,18 @@ async fn build_catalog() {
 
                     // We can build a catalog and start up the server and run a query
                     let restarted = RestartedServer::build_catalog_and_start(&table_dir).await;
-                    let batches = run_sql_until_non_empty(&restarted, sql, namespace.as_str())
-                        .with_timeout(Duration::from_secs(2))
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_cases/debug.rs:109]: I moved the timeout into `run_sql_until_non_empty` as I thought it made the test easier to read as the timeouts and deadline handling aren't inlined as much into the test logic

File: influxdb_iox/tests/end_to_end_cases/debug.rs [hunk 2]
```diff
@@ -215,6 +196,27 @@ impl RestartedServer {
             data_dir,
         }
     }
+
+    /// Runs the SQL query against this server, in a loop until
+    /// results are returned. Panics if the results are not produced
+    /// within a 5 seconds
+    async fn run_sql_until_non_empty(&self, sql: &str, namespace: &str) -> Vec<RecordBatch> {
+        let timeout = Duration::from_secs(5);
```
Comment (alamb) [inline influxdb_iox/tests/end_to_end_cases/debug.rs:204]: The real change of this PR are these two numbers

**Review by savage-engineer — APPROVED** [review]

**Comment by alamb** [comment]
lol -- the test failed on this PR so clearly this doesn't fix it. Marking as a draft while I look into it more

**Comment by alamb** [comment]
🤔  I heisenbug -- when I added some more debut output now it stops happening...

**Comment by alamb** [comment]
I can't for the life of me get this test to fail anymore though I do think it does on production. My next plan will be to add debugging information into the test, merge it in, and then study the logs

I'll make a new PR so we don't have the tortured history on this one

**Pull request closed without merging**